### ISSUE TITLE
make: *** No rule to make target 'dtbs_install'.  Stop.

### ISSUE THREAD

**Issue by doofin** [BredOS/sbc-pkgbuilds#8]
Hi, I try to run `makepkg -srAf` under `sbc-pkgbuilds/linux-rockchip-rkr3` but got this err `make: *** No rule to make target 'dtbs_install'.  Stop.`

I checked the kernel repo and found that there's no `dtbs_install` target in the Makefile. 
Maybe something is missing in the kernel repo or the pkgbuild file are not correct?

**Comment by Rippanda12** [comment]
Could you please go to `sbc-pkgbuilds/linux-rockchip-rkr3/src/linux-rockchip/` and tell me what branch it "defaulted" to?

**Comment by doofin** [comment]
it's rk6.1-rkr3

**Comment by Rippanda12** [comment]
For me `make -v` outputs:
```
GNU Make 4.4.1
Built for aarch64-unknown-linux-gnu
```

Are you building on x86 or ARM? If you are on x86 it wont work.

**Comment by doofin** [comment]
Yeah I'm on x86, but I use a cross tool chain. I managed to build the image successfully. Do you build this in a vm ?

**Comment by Rippanda12** [comment]
For kernel/any package we use a native arm machine, same for images. But since images dont require any compilation you can build them on any architecture.

**Comment by doofin** [comment]
I got those after compile : 
```bash
 > ls arch/arm64/boot/
dts  Image  Image.gz  install.sh  Makefile

 > file arch/arm64/boot/Image
arch/arm64/boot/Image: Linux kernel ARM64 boot executable Image, little-endian, 4K pages
```
I think now just need to get final steps to package them with makepkgs. But maybe I'm still missing something?

**Comment by Rippanda12** [comment]
> Do you use a rk3588 board to compile?
No we don't we use a high performance server to compile but with ccache it takes like 5 minutes to recompile first time compilation takes like 45-60 minutes on the rk3588.

**Comment by Rippanda12** [comment]
I don't know to be fair it works fine for us

**Comment by doofin** [comment]
> I don't know to be fair it works fine for us

Ok, let's see if @bill88t knows more, I think it will be very helpful if we can cross compile this

**Comment by dongioia** [comment]
For anyone hitting this when cross-compiling on x86_64: the root cause is that the kernel Makefile conditionally defines the `dtbs` / `dtbs_install` targets only when `arch/$(SRCARCH)/boot/dts/` exists. On an x86 host, `SRCARCH` defaults to `x86` (from `uname -m`), and `arch/x86/boot/dts/` doesn't exist — so the targets are never defined.

**Workaround**: pass `ARCH=arm64 CROSS_COMPILE=aarch64-linux-gnu-` to every `make` invocation. But since these PKGBUILDs are designed for native aarch64 builds, the cleanest approach is building inside an ARM64 environment (Docker container or QEMU user-mode):

```bash
# Example with Docker
docker run --rm -it -v $(pwd):/build arm64v8/archlinux:latest bash
# then install base-devel + deps and run makepkg inside
```

This matches how BredOS builds packages — natively on aarch64 hardware or in ARM64 containers.
